### DISCUSSION TITLE
로그인 http method GET vs POST

### DISCUSSION THREAD

**Discussion by hanull** [woowacourse-teams/2022-gong-seek#14] [General]
- POST
    - code 라는 데이터를 넘기기 때문에 POST가 적절하다고 생각. Get 메서드에 RequestBody로 넘기는 것을 제공하지 않는 경우도 있음 (버전별 상이)
    - 해당 요청에 의해서 Member가 생성되거나 수정될 수 있다. 즉, 서버의 리소스가 변화할 수 있기 때문에 POST가 적절하다고 생각.

**Comment by jurlring** [comment]
code라는 resource를 받고 token을 응답하는 매커니즘을 get으로 보기엔 조회하다와 거리가 있다고 생각되어서 일단 post를 적용하는 것이 좋은 것 같습니당

**Comment by Gyuchool** [comment]
post가 get보다는 보안적으로 조금 더 안전하므로 post를 통한 로그인이 좋아보입니다. code를 받는 다는게 oauth를 안쓰는 상황에서는 유저 정보를 받는경우라고 가정할 수 있어서 post로 안전하게 정보를 주고 받는게 좋은 것 같아요.

**Comment by brorae** [comment]
GET은 리소스를 가져올 때 사용합니다. OAuth를 사용하여 로그인할 때에는 토큰을 가져오긴 하지만, 새로운 리소스를 생성하고, 변경한다는 점에서 POST가 적절하다고 생각해요.
